src/muya/lib/parser/rules.js

```diff
@@ -21,7 +21,7 @@ export const inlineRules = {
   del: /^(~{2})(?=\S)([\s\S]*?\S)(\\*)\1/, // can nest
   auto_link: /^<(?:([a-zA-Z]{1}[a-zA-Z\d\+\.\-]{1,31}:[^ <>]*)|([a-zA-Z0-9.!#$%&'*+/=?^_`{|}~-]+@[a-zA-Z0-9](?:[a-zA-Z0-9-]{0,61}[a-zA-Z0-9])?(?:\.[a-zA-Z0-9](?:[a-zA-Z0-9-]{0,61}[a-zA-Z0-9])?)*))>/,
   // (extended www autolink|extended url autolink|extended email autolink) the email regexp is the same as auto_link.
-  auto_link_extension: /^(?:(www\.[a-z_-]+\.[a-z]{2,}(?::[0-9]{1,5})?(?:\/[\S]*)?)|(http(?:s)?:\/\/(?:[a-z0-9\-._~]+\.[a-z]{2,}|[0-9.]+|localhost|\[[a-f0-9.:]+\])(?::[0-9]{1,5})?(?:\/[\S]*)?)|([a-zA-Z0-9.!#$%&'*+/=?^_`{|}~-]+@[a-zA-Z0-9](?:[a-zA-Z0-9-]{0,61}[a-zA-Z0-9])?(?:\.[a-zA-Z0-9](?:[a-zA-Z0-9-]{0,61}[a-zA-Z0-9])?)*))(?=\s|$)/,
+  auto_link_extension: /^(?:(www\.[a-z_-]+\.[a-z]{2,}(?:\/[^\s:]*)?(?::[0-9]{1,5})?)|(http(?:s)?:\/\/(?:[a-z0-9\-._~]+\.[a-z]{2,}|[0-9.]+|localhost|\[[a-f0-9.:]+\])(?:\/[^\s:]*)?(?::[0-9]{1,5})?)|([a-zA-Z0-9.!#$%&'*+/=?^_`{|}~-]+@[a-zA-Z0-9](?:[a-zA-Z0-9-]{0,61}[a-zA-Z0-9])?(?:\.[a-zA-Z0-9](?:[a-zA-Z0-9-]{0,61}[a-zA-Z0-9])?)*))(?=\s|$|:)/,
```

Comment:
I can not understand why put `(?:\/[^\s:]*)?` in front of `(?::[0-9]{1,5})?`? I think `(?::[0-9]{1,5})?` is use to match the port like `:8080` and `(?:\/[^\s:]*)?` is use to match `path` and `query`?

Reply:
You are right, I was testing with the regex for a while and that was one of the areas I was adjusting. I believe for some reason I was thinking it was www.website.com/path:8080 and not www.website.com:8080/path/. I will push an update soon to resolve this.